### PR TITLE
Add support for publishing in Prometheus registry

### PR DESCRIPTION
**Pull request by marcocrasso**
Pull Request type
----

- [ ] Bugfix
- [X] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

Changes in this PR
----

This change-set would allow to publish in micrometer Prometheus registry using the Spring-Boot approach (i.e. actuator)-

Issue # N/A 

Alternatives considered
----

Before pushing this change-set I read contribs module, and oficial documentation. I found that the repo partially supported what I need, which was to make a Conductor server a Prometheus target using spring-boot standard approach (i.e. actuator). 
  
 I have enabled the test (i.e. removed the ignored annotation), in the test the Prometheus registry is programmatically injected because TestConfig declares a primary bean for it and the unit test uses @Autowired. This is what  happens in runtime when actuator is in the dependencies, and if the specific system property is satisfied, the registry is appended into the Spectator composite reg. 
 
 Test are passing, and the additions in docker folder allows running a Prometheus server which probes to consume everything: 

  
image

Your feedback is required, this is my first PR to this repo and I am sure that I am missing some etiquete or forget something. However, I found a couple of requests for this feature in past.
  
Please consider to review it and include it to main.

Build has failed with:
```
Execution failed for task ':sonarqube'.
224 actionable tasks: 191 executed, 33 up-to-date
> You're not authorized to run analysis. Please contact the project administrator.
```

### CODE REVIEW

**Comment by apanicker-nflx** [comment]
Thank you very much for your contribution. This will be of great help to the community especially for those who wish to use Prometheus with Conductor. 

Regarding the build failure, I have disabled sonarqube task for now since there seems to be a permissions issue.

File: contribs/src/test/java/com/netflix/conductor/contribs/metrics/TestConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
+
+@TestConfiguration
+public class TestConfig {
```
[the remark this answers is not in the document]
Comment (marcocrasso) [reply contribs/src/test/java/com/netflix/conductor/contribs/metrics/TestConfig.java]: Done, thanks for reviewing.

File: server/build.gradle [hunk 2]
```diff
@@ -19,6 +19,10 @@ dependencies {
 
     implementation 'org.springframework.boot:spring-boot-starter-log4j2'
     implementation 'org.apache.logging.log4j:log4j-web'
+
+    implementation 'org.springframework.boot:spring-boot-starter-actuator'
+
+    implementation group: 'com.netflix.spectator', name: 'spectator-reg-micrometer', version: '0.80.0'
```
[the remark this answers is not in the document]
Comment (marcocrasso) [reply server/build.gradle]: For sure, format applied and version fixed.

**Comment by marcocrasso** [comment]
Update:
I struggled with OOM during CI at HttpTaskTest test cases. After inspecting others PRs passing, I decided to skip these. 

On the contrary, OOM during CI at: 
 
- contribs/metrics/LoggingMetricsConfigurationTest.java
- contribs/metrics/PrometheusMetricsConfigurationTest.java

were solved, by loading the  spring-boot app context only once. 

If you know how to pass the skipped test in CI, I am really curious. Locally, they work well. 

Please review. M.

**Review by apanicker-nflx — APPROVED** [review]

**Review by aravindanr — APPROVED** [review]

**Pull request merged**